### PR TITLE
Migration script v2 set remoteWrite remoteTimeout to 5s

### PR DESCRIPTION
**Pull request by pmalek-sumo**
###### Description

Fill in your description here.

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Review by sumo-drosiek — COMMENTED** [review]
It would be nice to check if value is already set, but LGTM

**Comment by pmalek-sumo** [comment]
> It would be nice to check if value is already set, but LGTM

So I've tried a different approach with merging the temporary file that we work on with `remoteWrite` document (with specified `5s` `remoteTimeout`s) but the problem with this is that it might overwrite other values in that tree.

Maybe there's a way of doing this in `yq` but I'm not familiar with it. I'm also not sure if we can iterate over elements with `yq` and then take action in the script.

**Review by perk-sumo — COMMENTED** [review]
I think we shouldn't change this value at all.
Just check the user config and in case it is set we should display the INFO that we've changed the default value to 5.

**Comment by pmalek-sumo** [comment]
> I think we shouldn't change this value at all.
> Just check the user config and in case it is set we should display the INFO that we've changed the default value to 5.

Changed to just log if `remoteWrite` is set

**Comment by sumo-drosiek** [comment]
> Maybe there's a way of doing this in `yq` but I'm not familiar with it. I'm also not sure if we can iterate over elements with `yq` and then take action in the script.

It's possible in `yq`. We did it in upgrade-script for v1

File: deploy/helm/sumologic/upgrade-2.0.0.sh [hunk 1]
```diff
@@ -386,6 +386,18 @@ function add_new_scrape_labels_to_prometheus_service_monitors(){
...
+function kube_prometheus_stack_set_remote_write_timeout_to_5s() {
+  if [[ -z "$(yq r "${TEMP_FILE}" -- 'kube-prometheus-stack.prometheus.prometheusSpec.remoteWrite')" ]]; then
+    return
+  fi
+
+  if [[ -n "$(yq r "${TEMP_FILE}" -- 'kube-prometheus-stack.prometheus.prometheusSpec.remoteWrite.[*].remoteTimeout')" ]]; then
+    echo
+    info "kube-prometheus-stack.prometheus.prometheusSpec.remoteWrite.[*].remoteTimeout is set"
+    info "Please note that we've set it by default to 5s in 2.0.0"
+  fi
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/upgrade-2.0.0.sh:398]: We should inform if `remoteTImeout` is not set as well

File: tests/upgrade_v2_script/static/prometheus_remote_write_timeout.input.yaml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+              sourceLabels: [__name__]
+
+        - url: http://$(CHART).$(NAMESPACE).svc.cluster.local:9888/prometheus.metrics.applications.jmx
+          writeRelabelConfigs:
+            - action: keep
+              regex: (?:java_lang_(ClassLoading_(TotalL|Unl|L)oadedClassCount|Compilation_TotalCompilationTime|GarbageCollector_(Collection(Count|Time)|LastGcInfo_(GcThreadCount|duration|(memoryU|u)sage(After|Before)Gc_.*_used))|MemoryPool_(CollectionUsage(ThresholdSupported|_committed|_max|_used)|(Peak|)Usage_(committed|max|used)|UsageThresholdSupported)|Memory_((Non|)HeapMemoryUsage_(committed|max|used)|ObjectPendingFinalizationCount)|OperatingSystem_(AvailableProcessors|(CommittedVirtual|(Free|Total)(Physical|))MemorySize|(Free|Total)SwapSpaceSize|(Max|Open)FileDescriptorCount|ProcessCpu(Load|Time)|System(CpuLoad|LoadAverage))|Runtime_(BootClassPathSupported|Pid|Uptime|StartTime)|Threading_(CurrentThread(AllocatedBytes|(Cpu|User)Time)|(Daemon|Peak|TotalStarted|)ThreadCount|(ObjectMonitor|Synchronizer)UsageSupported|Thread(AllocatedMemory.*|ContentionMonitoring.*|CpuTime.*))))
+              sourceLabels: [__name__]
```
Comment (sumo-drosiek) [inline tests/upgrade_v2_script/static/prometheus_remote_write_timeout.input.yaml:117]: newline at EOF

File: deploy/helm/sumologic/upgrade-2.0.0.sh (hunk 1 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/upgrade-2.0.0.sh:398]: cc @perk-sumo

Comment (sumo-drosiek) [reply deploy/helm/sumologic/upgrade-2.0.0.sh:398]: or set it

Comment (perk-sumo) [reply deploy/helm/sumologic/upgrade-2.0.0.sh:398]: It is set by default in values.yaml, no need to add this into migrated conf file.

**Review by perk-sumo — APPROVED** [review]
👍

**Pull request merged**